org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CSARExporter.java

```diff
@@ -191,6 +194,187 @@ public void writeCSAR(TOSCAComponentId entryId, OutputStream out) throws Archive
+                }
+            } catch (Exception e) {
+                CSARExporter.LOGGER.error("Could not copy file to ZIP outputstream", e);
+            }
+			return;
+        }
+
+		Path tempDir = Files.createTempDirectory(WINERY_TEMP_DIR_PREFIX);
+		try {
+            Git git = Git
+                    .cloneRepository()
+                    .setURI(gitInfo.URL)
+                    .setDirectory(tempDir.toFile())
+                    .call();
+            git.checkout().setName(gitInfo.BRANCH).call();
+            String path = "artifacttemplates/"
+                    + Util.URLencode(((ArtifactTemplateId)ref.getParent().getParent()).getQName().getNamespaceURI())
+                    + "/"
+                    + ((ArtifactTemplateId)ref.getParent().getParent()).getQName().getLocalPart()
+                    + "/files/";
+			TArtifactTemplate template = Utils.getTArtifactTemplate((ArtifactTemplateDirectoryId) ref.getParent());
+			addWorkingTreeToArchive(zos, template, tempDir, path);
+        } catch (GitAPIException e) {
+            CSARExporter.LOGGER.error(String.format("Error while cloning repo: %s / %s", gitInfo.URL, gitInfo.BRANCH), e);
+        } finally {
+            deleteDirectory(tempDir);
+        }
+	}
+
+	/**
+	 * Adds a file to an archive
+	 * @param zos Output stream of the archive
+	 * @param ref Reference to the file that should be added to the archive
+	 * @param archivePath Path inside the archive to the file
+	 */
+	private void addFileToZipArchive(ArchiveOutputStream zos, RepositoryFileReference ref, String archivePath) {
+		try (InputStream is = Repository.INSTANCE.newInputStream(ref)) {
+            ArchiveEntry archiveEntry = new ZipArchiveEntry(archivePath);
+            zos.putArchiveEntry(archiveEntry);
+            IOUtils.copy(is, zos);
+        } catch (Exception e) {
+            CSARExporter.LOGGER.error("Could not copy file content to ZIP outputstream", e);
+        }
+	}
+
+	/**
+	 * Adds a dummy file to the archive
+	 * @param zos
+	 * @param transformer
+	 * @param ref
+	 * @param archivePath
+	 * @throws IOException
+	 */
+	private void addDummyRepositoryFileReferenceForGeneratedXSD(ArchiveOutputStream zos, Transformer transformer, DummyRepositoryFileReferenceForGeneratedXSD ref, String archivePath) throws IOException {
+		ArchiveEntry archiveEntry = new ZipArchiveEntry(archivePath);
+		zos.putArchiveEntry(archiveEntry);
+		CSARExporter.LOGGER.trace("Special treatment for generated XSDs");
+		Document document = ref.getDocument();
+		DOMSource source = new DOMSource(document);
+		StreamResult result = new StreamResult(zos);
+		try {
+            transformer.transform(source, result);
+        } catch (TransformerException e) {
+            CSARExporter.LOGGER.debug("Could not serialize generated xsd", e);
+        }
+	}
+
+	/**
+	 * Deletes a directory recursively
+	 * @param path Path to the directory that should be deleted
+	 */
+	private void deleteDirectory(Path path) {
```

Comment:
The files I need to delete are outside of the repository (`java.nio.Files.createTemporaryDirectory`) thus no Repository methods are accessible . Furthermore, java does not have a method for recursively deleting directories (ref: [SO thread](https://stackoverflow.com/a/35989142/1950267)).